aws_lambda_builders/workflows/nodejs_npm_esbuild/actions.py

```diff
@@ -79,6 +83,20 @@ def execute(self):
         args.append("--target={}".format(target))
         args.append("--outdir={}".format(self.artifacts_dir))
         try:
-            self.subprocess_esbuild.run(args, cwd=self.source_dir)
+            self.subprocess_esbuild.run(args, cwd=self.scratch_dir)
         except EsbuildExecutionError as ex:
             raise ActionFailedError(str(ex))
+
+    def _get_explicit_file_type(self, entry_point, entry_path):
+        if Path(entry_point).suffix:
+            if self.osutils.file_exists(entry_path):
+                return entry_point
+            raise ActionFailedError("entry point {} does not exist".format(entry_path))
+
+        if self.osutils.file_exists(get_typescript_file_type(entry_path)):
+            return get_typescript_file_type(entry_point)
+
+        if self.osutils.file_exists(get_javascript_file_type(entry_path)):
+            return get_javascript_file_type(entry_point)
```

Comment:
Do we need to do this refactoring and create `get_typescript_file_type` and `get_javascript_file_type`? I think this would be simpler if we don't need to use `get_typescript_file_type` and `get_javascript_file_type` elsewhere:
```suggestion
        for ext in [".ts", ".js"]:
            entry_path_with_ext = entry_path + ext
            if self.osutils.file_exists(entry_path_with_ext):
                return entry_path_with_ext
```